Composer/packages/client/src/store/action/lu.ts

```diff
@@ -9,25 +9,41 @@ import { ActionCreator, State } from '../types';
 
 import httpClient from './../../utils/httpUtil';
 import { ActionTypes } from './../../constants/index';
-import { fetchProject } from './project';
 import { setError } from './error';
 //remove editor's debounce and add it to action
-export const debouncedUpdateLu = debounce(async (store, id, projectId, content) => {
+export const debouncedUpdateLu = debounce(async (store, id, projectId, content, lastModified) => {
   try {
-    await httpClient.put(`/projects/${projectId}/luFiles/${id}`, { id, projectId, content });
+    const response = await httpClient.put(`/projects/${projectId}/luFiles/${id}`, {
+      id,
+      projectId,
+      content,
+      lastModified,
+    });
+    store.dispatch({
+      type: ActionTypes.UPDATE_TIMESTAMP,
+      payload: {
+        id: id,
+        type: 'lu',
+        lastModified: response.data.lastModified,
+      },
+    });
   } catch (err) {
     setError(store, {
       message: err.response && err.response.data.message ? err.response.data.message : err,
       summary: 'UPDATE LU ERROR',
     });
     //if update lu error, do a full refresh.
-    fetchProject(store);
+    // fetchProject(store);
```

Comment:
what's the plan here?

Reply:
Right now, the call to setError causes a popup to appear that tells the user to reload the page to get the latest version of the content.

I think we could specialize that to catch the 409 error that is returned in this instance and show a dialog that *actually does do the fetchProject* for the user instead of having them manually reload the page. This would also take care of the static error message being priovided by the server (comments below)